consensus/istanbul/backend/backend.go

```diff
@@ -497,9 +498,27 @@ func (sb *Backend) Commit(proposal istanbul.Proposal, aggregatedSeal types.Istan
 		return nil
 	}
 
-	if sb.broadcaster != nil {
-		sb.broadcaster.Enqueue(fetcherID, block)
+	// If no cached result, we import block through fetcher, which requires processing the block one more time.
+	// Else if cache hit, we directly write the block.
+	if result == nil {
+		sb.logger.Warn("state process cache miss, enqueuing the block to fetcher", "number", block.Number(), "hash", block.Hash())
+		if sb.broadcaster != nil {
```

Comment:
I think this log is slightly out of place
```suggestion
		if sb.broadcaster != nil {
				sb.logger.Warn("state process cache miss, enqueuing the block to fetcher", "number", block.Number(), "hash", block.Hash())
```